documentation/release_notes.markdown

```diff
@@ -16,7 +16,46 @@ For release notes on versions of Puppet Server prior to Puppet Server 5, see [do
 
 Released June 27, 2017.
 
-This is a major release of Puppet Server.
+This is a major release of Puppet Server, and corresponds with the major release of Puppet 5.0, which also includes many changes and new features relevant to Puppet Server users. 
+
+### Platform changes
+
+Puppet Server 5.0 packages are available for RHEL 6 and 7, Debian 8 (Jessie), Ubuntu 16.04 (Xenial), and SLES 12 SP1.
+
+Puppet Server 5.0 is built with JDK 8, and therefore cannot run on a Java 7 runtime. Server 5.0 packages now depend exclusively upon the `openjdk-8-jre-headless` package. Since Ubuntu Precise and Trusty, and versions of Debian prior to 8 (Jessie), do not distribute that package, we do not provide Puppet Server packages for those operating systems.
+
+For SLES 12 SP1, install the `jessie-backports` repository to add access to Java 8. For details, see [Installing From Packages](./install_from_packages.markdown).
+
+-   [SERVER-1738](https://tickets.puppetlabs.com/browse/SERVER-1738)
+-   [SERVER-1741](https://tickets.puppetlabs.com/browse/SERVER-1741)
+-   [SERVER-1800](https://tickets.puppetlabs.com/browse/SERVER-1800)
+
+### Upgrading from previous versions
+
+Consult the [Puppet 5.0 release notes](https://docs.puppet.com/puppet/5.0/release_notes.html) for more information about important new features and breaking changes. Some especially relevant Puppet 5.0 changes are also noted below.
+
+#### Deprecation: Puppet 5 no longer writes YAML node caches
+
+As of Puppet 5.0, Puppet no longer writes node YAML files to its cache by default. This cache has been used in workflows where external tooling needs a list of nodes, but PuppetDB is now the preferred source of node information.
+
+To retain the Puppet 4.x behavior, add the [`puppet.conf`](./configuration.markdown) setting `node_cache_terminus = write_only_yaml`. Note that `write_only_yaml` is deprecated, and users are encouraged to migrate to PuppetDB in order to retrieve node information.
+
+-   [SERVER-1819](https://tickets.puppetlabs.com/browse/SERVER-1819)
+-   [PUP-6060](https://tickets.puppetlabs.com/browse/PUP-6060)
+
+#### Deprecation/New Feature: Puppet 5 uses JSON serialization by default
+
+Previous versions of Puppet exclusively used PSON, our vendored version of `pure_json`, for serializing communication between agents and masters. Testing showed that PSON serialization's performance was significantly worse than using JSON, and JSON adds opportunities for easier and better interoperability with other tools and programming languages.
+
+As part of this, Puppet Server 5.0 requests JSON over PSON from Puppet 5.0 agents by default, and consumes JSON facts and reports provided by Puppet 5.0 agents.
```

Comment:
Technically speaking, the Puppet Server never requests JSON/PSON from the agent.  The communication is all driven from the agent instead.

Puppet 5 agents will, by default, request that the server provide responses to it in application/json or text/pson - whereas Puppet 4 agents would only request text/pson.  Puppet 5 agents will also, by default, encode parts of their request bodies (the full payload for a reports request and the facts parameter in a catalog request) in application/json - whereas Puppet 4 agents would try to use text/pson for these request bodies instead.

Puppet Server 5 has been enhanced to:

* Produce application/json responses when a Puppet 5 agent requests them.
* Consume application/json request bodies sent from the Puppet 5 agent.

Note that the Puppet 5 agent will attempt to downgrade to requesting / sending only text/pson (and not application/json) when it detects that the server it is talking to is not capable of supporting application/json - for example, a 2.x version of Puppet Server.

Does this make sense?